components/automate-ui/src/app/entities/projects/project.reducer.ts

```diff
@@ -96,6 +117,9 @@ export function projectEntityReducer(
     case ProjectActionTypes.UPDATE_FAILURE:
       return set(UPDATE_STATUS, EntityStatus.loadingFailure, state);
 
+    case ProjectActionTypes.GET_APPLY_RULES_STATUS_SUCCESS:
+      return set(APPLY_RULES_STATUS, mapKeys(key => camelCase(key), action.payload), state);
```

Comment:
What are you camel-casing here? Seems like the only field would be the state, and that uses snake case, no?

Reply:
The GET response payload (and all other API responses) returns as snakecase:
```
{
  "state": "not_running",
  "estimated_time_complete": "0001-01-01T00:00:00Z",
  "percentage_complete": 1,
  "failed": false,
  "failure_message": ""
}
```

However I have [the interface](https://github.com/chef/automate/blob/0c43a66cc0f46bd36c40fab549a08e7e5d352f21/components/automate-ui/src/app/entities/projects/project.reducer.ts#L11-L17) set up as camelcase (since its JS and thats what we use everywhere):
```
interface ApplyRulesStatus {
  state: ApplyRulesStatusState;
  estimatedTimeComplete: string;
  percentageComplete: number;
  failed: boolean;
  failureMessage: string;
}
```

As for camel-casing here in the reducer specifically, I could've also done it in the `project.requests` method that gets the response or in the effect that handles the response. I just went w/ doing it in the reducer; I'd be fine doing it elsewhere (or not at all and instead change the interface to snakecase).

I wish we had chosen camelcase for our JSON APIs 😐

Reply:
Ah, I was thinking about the data, not the field names. 👍